<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>656</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1002</width>
    <height>656</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1002</width>
    <height>656</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Кодировщик</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTextEdit" name="InitialText_TE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>471</width>
      <height>181</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="placeholderText">
     <string>Исходный текст</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="ResultText_TE">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>60</y>
      <width>471</width>
      <height>181</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="placeholderText">
     <string>Результат</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Alphabet_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>471</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="dragEnabled">
     <bool>false</bool>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string>Алфавит</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="InitialTextFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>37</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с исходным текстом</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="AlphabetFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>257</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с алфавитом</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Frequency_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>471</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string>Вероятности</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="FrequencyFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>327</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с вероятностями</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Codes_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>471</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string>Коды символов</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="CodesFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>397</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с кодами символов</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ResultTextFilename_LE">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>37</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с результатом</string>
    </property>
   </widget>
   <widget class="QPushButton" name="InitialTextRead_B">
    <property name="geometry">
     <rect>
      <x>192</x>
      <y>38</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Прочитать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ResultTextWrite_B">
    <property name="geometry">
     <rect>
      <x>728</x>
      <y>38</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Записать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AlphabetRead_B">
    <property name="geometry">
     <rect>
      <x>192</x>
      <y>258</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Прочитать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FrequencyRead_B">
    <property name="geometry">
     <rect>
      <x>192</x>
      <y>328</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Прочитать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AlphabetWrite_B">
    <property name="geometry">
     <rect>
      <x>273</x>
      <y>258</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Записать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FrequencyWrite_B">
    <property name="geometry">
     <rect>
      <x>273</x>
      <y>328</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Записать</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>258</y>
      <width>471</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="LBL_1">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>10</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Длина кодового слова:</string>
     </property>
    </widget>
    <widget class="QLabel" name="LBL_2">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>40</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Энтропия:</string>
     </property>
    </widget>
    <widget class="QLabel" name="LBL_3">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>70</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Избыточность:</string>
     </property>
    </widget>
    <widget class="QLabel" name="LBL_4">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>100</y>
       <width>171</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Неравенство Крафта:</string>
     </property>
    </widget>
    <widget class="QLabel" name="KraftInequality_LBL">
     <property name="geometry">
      <rect>
       <x>195</x>
       <y>100</y>
       <width>221</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>нет</string>
     </property>
    </widget>
    <widget class="QLabel" name="Redundancy_LBL">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>70</y>
       <width>171</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="Entropy_LBL">
     <property name="geometry">
      <rect>
       <x>105</x>
       <y>40</y>
       <width>161</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="WordAverageLength_LBL">
     <property name="geometry">
      <rect>
       <x>204</x>
       <y>10</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="CodesCreate_B">
    <property name="geometry">
     <rect>
      <x>192</x>
      <y>398</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Получить</string>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="CodesWrite_B">
    <property name="geometry">
     <rect>
      <x>273</x>
      <y>398</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Записать</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>390</y>
      <width>431</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Хаффман</string>
    </property>
    <widget class="QPushButton" name="Encode_B">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Зашифровать</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Decode_B">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Расшифровать</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Clear_B">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Очистить 
таблицу</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>561</y>
      <width>431</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Хэмминг</string>
    </property>
    <widget class="QPushButton" name="HammingEncode_B">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Зашифровать</string>
     </property>
    </widget>
    <widget class="QPushButton" name="HammingDecode_B">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Расшифровать</string>
     </property>
    </widget>
    <widget class="QPushButton" name="HammingClear_B">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Очистить 
таблицу</string>
     </property>
    </widget>
   </widget>
   <widget class="QTextEdit" name="Matrix_TE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>510</y>
      <width>471</width>
      <height>121</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="placeholderText">
     <string>Проверочная матрица</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="MatrixFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>466</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с матрицей</string>
    </property>
   </widget>
   <widget class="QPushButton" name="MatrixRead_B">
    <property name="geometry">
     <rect>
      <x>192</x>
      <y>488</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Прочитать</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="MatrixSizesFilename_LE">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>487</y>
      <width>171</width>
      <height>24</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>Файл с размерами</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>490</y>
      <width>471</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="LBL_5">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>10</y>
       <width>161</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Количество ошибок:</string>
     </property>
    </widget>
    <widget class="QLabel" name="LBL_6">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>40</y>
       <width>161</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Ошибки в позициях:</string>
     </property>
    </widget>
    <widget class="QLabel" name="ErrsPos_LBL">
     <property name="geometry">
      <rect>
       <x>184</x>
       <y>40</y>
       <width>281</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="ErrCount_LBL">
     <property name="geometry">
      <rect>
       <x>184</x>
       <y>10</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1002</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
